### PR TITLE
Unsupported if wrong image used on virtualization

### PR DESCRIPTION
**Pull request by mdegat01**
## Proposed change

Home Assistant OS has a specific image for use on virtualized systems called `ova`. It is possible to install the `generic-x86-64` and people do but then they report issues when expected options like guest tools are missing. This lets them know they have installed the wrong image and the repair will direct them to the correct one for their environment.

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (which adds functionality to the supervisor)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue:
- Link to documentation pull request:
     - Translation strings for repair: https://github.com/home-assistant/core/pull/113882
     - Unsupported reason page: https://github.com/home-assistant/home-assistant.io/pull/31971
     - API Doc: https://github.com/home-assistant/developers.home-assistant/pull/2117
- Link to cli pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] Local tests pass. **Your PR cannot be merged unless tests pass**
- [ ] There is no commented out code in this PR.
- [ ] I have followed the [development checklist][dev-checklist]
- [ ] The code has been formatted using Ruff (`ruff format supervisor tests`)
- [ ] Tests have been added to verify that the new code works.

If API endpoints of add-on configuration are added/changed:

- [ ] Documentation added/updated for [developers.home-assistant.io][docs-repository]

[dev-checklist]: https://developers.home-assistant.io/docs/en/development_checklist.html
[docs-repository]: https://github.com/home-assistant/developers.home-assistant

### CODE REVIEW

**Comment by mdegat01** [comment]
@agners do you want to write the unsupported page for this one? We'll need a new page at `https://www.home-assistant.io/more-info/virtualized_image` (assuming we're ok with that identifier for this issue)

File: supervisor/resolution/evaluations/virtualization_image.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    @property
+    def states(self) -> list[CoreState]:
+        """Return a list of valid states when this evaluation can run."""
+        return [CoreState.STARTUP]
+
+    async def evaluate(self):
+        """Run evaluation."""
+        if not self.sys_os.available:
+            return False
+        return self.sys_dbus.systemd.virtualization and self.sys_os.board != "ova"
```
Comment (sairon) [inline supervisor/resolution/evaluations/virtualization_image.py:36]: I _think_ it's valid for `generic-aarch64` to run in a VM too.

File: supervisor/resolution/evaluations/virtualization_image.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    @property
+    def states(self) -> list[CoreState]:
+        """Return a list of valid states when this evaluation can run."""
+        return [CoreState.STARTUP]
+
+    async def evaluate(self):
+        """Run evaluation."""
+        if not self.sys_os.available:
+            return False
+        return self.sys_host.info.virtualization and self.sys_os.board not in {
```
Comment (agners) [inline supervisor/resolution/evaluations/virtualization_image.py:36]: I was thinking to use the Chassis property and check for `vm` here. But checking virtualization is most likely largely equivalent. It seems that if Chassis can't be read through DMI BIOS interface, it falls back to the virtualization detection (see https://github.com/systemd/systemd/blob/v252/src/hostname/hostnamed.c#L283-L286).

I guess with this we also mark container as unsupported, which most likely make sense. I don't think it is reasonable or possible to run full HAOS in a container :sweat_smile:

File: supervisor/core.py [hunk 3]
```diff
@@ -194,6 +194,9 @@ async def start(self):
         """Start Supervisor orchestration."""
         self.state = CoreState.STARTUP
 
+        # Evaluate the system
+        await self.sys_resolution.evaluate.evaluate_system()
```
Comment (agners) [inline supervisor/core.py:198]: Hm, is this additional evaluation really necessary? It seems to me that the one at the end of setup just above should be enough? Or maybe we can drop that one instead?

File: supervisor/resolution/evaluations/virtualization_image.py (hunk 2 above)
Comment (mdegat01) [reply supervisor/resolution/evaluations/virtualization_image.py:36]: So is this good then or should I change to chassis? Or check both?

Comment (agners) [reply supervisor/resolution/evaluations/virtualization_image.py:36]: Either way should work, I think it is fine to use the virtualization property.

**Review by agners — APPROVED** [review]

**Pull request merged**